src/creature.cpp

```diff
@@ -527,26 +528,26 @@ void Creature::onDeath()
 	}
 
 	if (droppedCorpse) {
-		g_game.removeCreature(this, false);
+		g_game.removeCreature(getCreature(), false);
 	}
 }
 
-bool Creature::dropCorpse(Creature* lastHitCreature, Creature* mostDamageCreature, bool lastHitUnjustified,
-                          bool mostDamageUnjustified)
+bool Creature::dropCorpse(std::shared_ptr<Creature> lastHitCreature, std::shared_ptr<Creature> mostDamageCreature,
+                          bool lastHitUnjustified, bool mostDamageUnjustified)
 {
 	if (!lootDrop && getMonster()) {
 		if (master) {
 			// scripting event - onDeath
 			const CreatureEventList& deathEvents = getCreatureEvents(CREATURE_EVENT_DEATH);
 			for (CreatureEvent* deathEvent : deathEvents) {
-				deathEvent->executeOnDeath(this, nullptr, lastHitCreature, mostDamageCreature, lastHitUnjustified,
-				                           mostDamageUnjustified);
+				deathEvent->executeOnDeath(getCreature(), nullptr, lastHitCreature, mostDamageCreature,
+				                           lastHitUnjustified, mostDamageUnjustified);
 			}
 		}
 
 		g_game.addMagicEffect(getPosition(), CONST_ME_POFF);
 	} else {
-		Item* splash;
+		std::shared_ptr<Item> splash;
```

Comment:
Could this be `auto`
```suggestion
		auto splash = std::shared_ptr<Item>(nullptr);
```